client/js/compute.js

```diff
@@ -0,0 +1,578 @@
+      call: function(v, x, y) { v[x] = y; return v }
+    },
+    {
+      name: 'from',
+      desc: 'converts and returns an array of the given string or object',
+      sample: 'var a = from ${x}',
+      call: function(v, x) { return v = Array.from(x) }
+    },
+    {
+      name: 'isArray',
+      desc: 'returns true if the object is an array, and false if not',
+      sample: 'var a = isArray ${x}',
+      call: function(v, x) { return v = Array.isArray(x) }
+    },
+    {
+      name: 'concatArray',
+      desc: 'returns an array containing the elements of the two joined arrays',
+      sample: 'var a = ${x} concatArray ${y}',
+      call: function(v, x, y) { return v = x.concat(y) }
+    },
+    {
+      name: 'pop',
+      desc: 'returns an array after removing its last element',
+      sample: 'var a = ${x} pop',
+      call: function(v, x) { return v = x.pop() }
+    },
+    {
+      name: 'reverse',
+      desc: 'returns an array after reversing the order of its elements',
+      sample: 'var a = ${x} reverse',
+      call: function(v, x) { return v = x.reverse() }
+    },
+    {
+      name: 'shift',
+      desc: 'returns an array after removing its first element',
+      sample: 'var a = ${x} shift',
+      call: function(v, x) { return v = x.shift() }
+    },
+    {
+      name: 'sort',
+      desc: 'returns an array after sorting its elements alphabetically',
+      sample: 'var a = ${x} sort',
+      call: function(v, x) { return v = x.sort() }
+    },
+    {
+      name: 'numericSort',
+      desc: 'returns an array after sorting its elements numerically',
+      sample: 'var a = ${x} numericSort',
+      call: function(v, x) { return v = x.sort((a, b) => a.toString().localeCompare(b, undefined, {numeric: true, ignorePunctuation: true})) }
+    },
+    {
+      name: 'findIndex',
+      desc: 'returns the index of the first element in the array that matches the given value y',
+      sample: 'var a = ${x} findIndex ${y}',
+      call: function(v, x, y) { return v = x.findIndex(y) }
+    },
+    {
+      name: 'join',
+      desc: 'returns the array x as a combined string with values separated by an optional separator y, defaulting to comma',
+      sample: 'var a = ${x} join ${y}',
+      call: function(v, x, y) { return v = (y !== undefined) ? x.join(y) : x.join() }
+    },
+    {
+      name: 'slice',
+      desc: 'returns elements between start index y an end index z of an array (or characters of a string), excluding the end (unless z is omitted)',
+      sample: 'var a = ${x} slice ${y} ${z}',
+      call: function(v, x, y, z) { return v = (z !== undefined) ? x.slice(y, z) : x.slice(y) }
+    },
+    {
+      name: 'splice',
+      desc: 'removes elements starting from index x from the array in variable a (optionally limited to only y elements)',
+      sample: 'var a = splice ${x} ${y}',
+      call: function(v, x, y) { if(y !== undefined) {v.splice(x, y)} else {v.splice(x)}; return v }
```

Comment:
Why not use `z` for parameter 3?

Reply:
Yes, we could allow users to remove elements and add a single element (z), but I wanted to keep this straightforward at first. Instead of supporting just a single addition, I was wondering if we should add some kind of `insert` functions instead, which would allow to insert a single value or an array into another array?
```
var firstArray = insert ${position} ${someStringOrNumber}
var firstArray = insertArray ${position} ${secondArray}
```

Reply:
so basically: v.splice($x, 0, $y), but call it insert?

Reply:
Oh, I see, I thought the third parameter was an array. We could just implement it that way? Basically your `insertArray` idea.

Reply:
I may be misunderstanding and/or wrong, but to me it seems like for insertArray to insert multiple elements (e.g multiple strings or multiple numbers) instead of inserting an array into another array, the array would need to be converted to a string that would then be the $y value of  v.splice($x, 0, $y).

Reply:
I believe this could be implemented with the spread operator (`.splice(position, 0, ...someArray)`). I can probably make `insert` work for both strings and arrays by first checking for the argument's type (like the `in` function). And the counterpart `remove` could then also be implemented to work on both strings and arrays. The question is: do we then also need `splice` as a combined remove & insert?

Reply:
We should probably  apply the same syntax of `unshift` and `push` for `insert` and `remove`, so `var b = insert 1 'foo'` and the same handling of the returned value (i.e. returning the resulting array instead of the removed elements) for consistency.

Reply:
I see.
Personally, I don't see much difference between keeping splice as a combined remove and insert and not.  I can't think of a strong argument either way.  Things I might be concerned about is if keeping it would confuse some people who aren't familiar with splice = or if not having it would confuse some people who are familiar with how splice works.